packages/babel-plugin/src/index.ts

```diff
@@ -0,0 +1,83 @@
+import type {PluginObj, NodePath} from '@babel/core';
+import type { BabelAPI } from '@babel/helper-plugin-utils';
+import type {PluginOptions, PluginState} from './types';
+
+import { generate as babelGenerate } from '@babel/generator';
+import { types as babelTypes } from '@babel/core';
+import { declare } from "@babel/helper-plugin-utils";
+import { PRAGMA, REPLACEMENT } from './identifiers.json';
+import { getIsolatedArrowFunction } from './helpers/transformArrowFunction';
+import validatePragmaUse from './helpers/validatePragmaUse';
+import validateCallExpressionAndGetArrowFunction from './helpers/validateCallExpressionAndGetArrowFunction';
+
+export default declare<
+  PluginOptions, 
+  // @ts-expect-error There's an issue with DefinitelyTyped here https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72756
+  PluginObj<PluginState>
+>((api: BabelAPI, options: PluginOptions, dirname: string): PluginObj<PluginState> => {
+  api.assertVersion(7);
+
+  return {
+    visitor: {
+      Program: {
+        enter(programPath, state) {
+          // TODO: Check if PRAGMA was imported from correct module
```

Comment:
I'm still not sure what PRAGMA is in your code, but if it's the helper function that will wrap the compiled function, it doesn't need to be pre-imported.

Reply:
Yeah it's the helper function, just something to rename easily, right now it's `$identifier`